### ISSUE TITLE
Existing implementations?

### ISSUE THREAD

**Issue by pmjones** [payload-interop/payload-interop#4]
What are the existing Domain Payload implementations?

**Comment by pmjones** [comment]
https://github.com/auraphp/Aura.Payload_Interface

https://github.com/auraphp/Aura.Payload

**Comment by shochdoerfer** [comment]
This is the interface we use:
https://github.com/bitExpert/adroit/blob/master/src/bitExpert/Adroit/Domain/Payload.php

And this is the implementation in our ADR "framework":
https://github.com/bitExpert/adrenaline/blob/master/src/bitExpert/Adrenaline/Domain/DomainPayload.php

**Comment by pmjones** [comment]
Via Packagist search for "domain payload", also these (not including deprecated/abandoned/missing):

- https://github.com/JoeBengalen/Payload

- https://github.com/dashifen/domain/tree/master/src/Payload

- https://github.com/equip/adr/blob/master/src/PayloadInterface.php

- ~~https://github.com/gpslab/payload~~

- https://github.com/geggleto/Payload/blob/master/src/Payload.php

**Comment by pmjones** [comment]
@JoeBengalen @geggleto @dashifen @gpslab you may (or may not!) be interested in providing input and advice on this project.

**Comment by geggleto** [comment]
I had written that so long ago; I will gladly help out :)

**Comment by pmjones** [comment]
Adding https://github.com/perfect-oblivion/payload by @devcircus to the list

**Comment by pmjones** [comment]
It occurs to me that the only portion that needs interoperability is the "reading" portion of the Payload. Since the Domain is what creates the Payload, and Domains are typically very specific in their nature, the creation methods need not be standardized.

Likewise, the specific status codes themselves need not be standardized either, since they are going to be Domain-related. (I could see some room for generic statuses like SUCCESS, FAILURE, etc., but it would not be strictly necessary.)

**Comment by dashifen-gu** [comment]
I'm not sure I have anything else to add, here, really.  I was mostly messing around anyway, but I wish you all the best of luck.

**Comment by pmjones** [comment]
Adding https://github.com/priskz/payload by @priskz to the list: read-only, with `getStatus() : mixed` and `getData() : mixed` as the readable signature.

**Comment by priskz** [comment]
> Adding https://github.com/priskz/payload by @priskz to the list: read-only, with `getStatus() : mixed` and `getData() : mixed` as the readable signature.

Hey @pmjones ! Thank you for your contributions to the open source community. I took a lot of inspiration from your ADR pattern when I was putting together my implementation of it for my own projects. I ended up calling it SO-RAD (Service Oriented - Responder Action Domain). I have some documentation here: https://docs.google.com/presentation/d/13_zydQ-o8nxTzDBRjKx4dPJQNXWDaIaKODx5VeBLg1A/edit?usp=sharing . I think it could be beneficial for anyone interested in ADR & DDD patterns.
